src/wp-includes/media.php

```diff
@@ -5660,11 +5658,18 @@ function wp_getimagesize( $filename, ?array &$image_info = null ) {
 		return $info;
 	}
 
+	$image_mime_type = wp_get_image_mime( $filename );
+
+	// Not an image?
+	if ( ! $image_mime_type ) {
```

Comment:
Nitpicky from my side, but why not be strict here?

```suggestion
	if ( false === $image_mime_type ) {
```